test/test_allocator.cpp

```diff
@@ -0,0 +1,96 @@
+// Copyright 2024 Man Group Operations Limited
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "doctest/doctest.h"
+
+#include <memory>
+#include <memory_resource>
+#include <numeric>
+#include <vector>
+
+#include "sparrow/allocator.hpp"
+
+TEST_SUITE("any_allocator")
+{
+    TEST_CASE_TEMPLATE_DEFINE("value semantic", A, value_semantic_id)
+    {
+        SUBCASE("constructor")
+        {
+            {
+                sparrow::any_allocator<int> a;
+            }
+            {
+                A alloc;
+                sparrow::any_allocator<typename A::value_type> a(alloc);
+            }
+        }
+
+        SUBCASE("copy semantic")
+        {
+            using value_type = typename A::value_type;
+
+            A alloc;
+            sparrow::any_allocator<value_type> a(alloc);
+            sparrow::any_allocator<value_type> b(a);
+            CHECK(a == b);
+
+            sparrow::any_allocator<value_type> c(alloc);
+            b = c;
+            CHECK(b == c);
+        }
```

Comment:
Side comment, in case: I dont know if it's possible to check with allocators, but usually checking copy operations implies checking that modifying one of the copy will not affect the others. Not sure it makes sense here, because an allocator could be designed to be an access point to an Arena or Pool.

Reply:
Indeed it does not really make sense here, as `std::allocator` is stateless, and you can not change the resource pointer of a `polymorphic_allocator` when it's built.